modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_impl.ftl

```diff
@@ -76,6 +84,32 @@ public abstract class Base${schemaName}ResourceImpl implements ${schemaName}Reso
 		public ${javaMethodSignature.returnType} ${javaMethodSignature.methodName}(${freeMarkerTool.getResourceParameters(javaMethodSignature.javaMethodParameters, openAPIYAML, javaMethodSignature.operation, true)}) throws Exception {
 			<#if stringUtil.equals(javaMethodSignature.returnType, "boolean")>
 				return false;
+			<#elseif stringUtil.equals(javaMethodSignature.methodName, "put" + schemaName + "Permission")>
+
+				PermissionChecker permissionChecker =
+					PermissionThreadLocal.getPermissionChecker();
+
+				String resourceName = get${schemaName}ResourceName();
+
+				if (!permissionChecker.hasPermission(
+						0, resourceName, 0, ActionKeys.PERMISSIONS)) {
+
+					return;
+				}
+
+				resourcePermissionLocalService.updateResourcePermissions(
+					contextCompany.getCompanyId(), 0, resourceName,
+					String.valueOf(structuredContentId),
```

Comment:
@nhpatt structuredContentId is hard-coded by mistake, should be the argument passed to the method.

Reply:
Brian noticed it and fixed it later

Reply:
@nhpatt strange, it seems he hasn't pushed to remote yet?
https://github.com/brianchandotcom/liferay-portal/blob/master/modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_impl.ftl#L96

Reply:
You are right, he changed another reference but there were more. Thanks!

Reply:
fixed by https://github.com/brianchandotcom/liferay-portal/pull/83562